lightning/src/ln/channel.rs

```diff
@@ -1652,6 +1652,9 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 		if msg.amount_msat > self.channel_value_satoshis * 1000 {
 			return Err(ChannelError::Close("Remote side tried to send more than the total value of the channel"));
 		}
+		if msg.amount_msat == 0 {
+			return Err(ChannelError::Ignore("Remote side tried to send a 0-msat HTLC"));
```

Comment:
I dont think this is right? We should either ::Close() or we should add the HTLC and fail it backwards as soon as they commit.

Reply:
You right, should fail the channel IIRC the spec, will update